rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -841,12 +841,245 @@ void MCAPStorage::ensure_rosdistro_metadata_added()
+  std::vector<rosbag2_storage::TopicMetadata> get_all_topics_and_types() override;
+  void get_all_message_definitions(
+          std::vector<rosbag2_storage::MessageDefinition> & definitions) override;
+
+  /** ReadOnlyInterface **/
+  void set_filter(const rosbag2_storage::StorageFilter & storage_filter) override;
+  void reset_filter() override;
+  #ifdef ROSBAG2_STORAGE_MCAP_OVERRIDE_SEEK_METHOD
+  void seek(const rcutils_time_point_value_t & time_stamp) override;
+  #else
+  void seek(const rcutils_time_point_value_t & time_stamp);
+  #endif
+
+  /** ReadWriteInterface **/
+  uint64_t get_minimum_split_file_size() const override;
+
+  /** BaseWriteInterface **/
+  void write(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> msg) override;
+  void write(
+          const std::vector<std::shared_ptr<const rosbag2_storage::SerializedBagMessage>> & msg) override;
+  void create_topic(const rosbag2_storage::TopicMetadata & topic,
+                    const rosbag2_storage::MessageDefinition & message_definition) override;
+  void remove_topic(const rosbag2_storage::TopicMetadata & topic) override;
+  #ifdef ROSBAG2_STORAGE_MCAP_HAS_UPDATE_METADATA
+  void update_metadata(const rosbag2_storage::BagMetadata &) override;
+  #endif
+
+private:
+   std::shared_ptr<MCAPStorage> storage_;
+   std::string uri_;
+   rosbag2_storage::storage_interfaces::IOFlag io_flag_ = rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE;
+   rosbag2_storage::BagMetadata metadata_;
+   std::vector<std::pair<rosbag2_storage::TopicMetadata,rosbag2_storage::MessageDefinition>> topics_;
+   rosbag2_storage::StorageOptions storage_options_;
+   uint64_t current_storage_size_;
+};
+
+MCAPStorageSnapshot::MCAPStorageSnapshot():storage_(nullptr),current_storage_size_(0)
+{
+}
+
+MCAPStorageSnapshot::~MCAPStorageSnapshot()
+{
+}
+
+/** BaseIOInterface **/
+#ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
+void MCAPStorageSnapshot::open(const rosbag2_storage::StorageOptions & storage_options,
+                       rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  (void)io_flag;
+  storage_options_ = storage_options;
+  uri_ = storage_options.uri;
+}
+#endif
+
+void MCAPStorageSnapshot::open(const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  uri_ = uri;
+  //this class is used only for READ_WRITE
+  (void)io_flag;
+}
+
+/** BaseInfoInterface **/
+rosbag2_storage::BagMetadata MCAPStorageSnapshot::get_metadata()
+{
+  return metadata_;
+}
+
+std::string MCAPStorageSnapshot::get_relative_file_path() const
+{
+  return uri_;
+}
+
+uint64_t MCAPStorageSnapshot::get_bagfile_size() const
+{
+  return current_storage_size_;
+}
+
+std::string MCAPStorageSnapshot::get_storage_identifier() const
+{
+  return "mcap_snapshot";
+}
+
+/** BaseReadInterface **/
+#ifdef ROSBAG2_STORAGE_MCAP_HAS_SET_READ_ORDER
+bool MCAPStorageSnapshot::set_read_order(const rosbag2_storage::ReadOrder & read_order)
+{
+  (void)read_order;
```

Comment:
Don't `(void)` variables. Declare them as comments in the function definition:

```c++
bool MCAPStorageSnapshot::set_read_order(const rosbag2_storage::ReadOrder & /*read_order*/)
```

Reply:
fixed